src/main/java/ua/com/javarush/oleksandr/reddit/redditcloneabstract/controller/SubredditController.java

```diff
@@ -31,60 +28,56 @@ public class SubredditController {
 
     private final SubredditService subredditService;
     private final SubredditMapper subredditMapper;
+    private final ErrorHandlerService errorHandlerService;
+    private final MessageSource messageSource;
 
     @PostMapping
     public ResponseEntity<Void> createSubreddit(@RequestBody @Valid SubredditRequestDTO subredditRequestDTO,
-                                                               BindingResult bindingResult) {
-        // TODO: implement subredditValidator.validate(subredditDTO);
-        if (bindingResult.hasErrors()) {
-            // TODO: implement throw exception
-            log.warn("Invalid subreddit data: {}", bindingResult.getAllErrors());
-            throw new ResponseStatusException(HttpStatus.BAD_REQUEST, "Invalid subreddit data");
-        }
+                                                BindingResult bindingResult) {
 
-        Subreddit subreddit = subredditMapper.subredditRequestDtoToSubreddit(subredditRequestDTO);
+        errorHandlerService.handle(bindingResult, SubredditCreateException::new);
```

Comment:
In the save method there is already processing and using handle, I think this line is unnecessary.